mesonbuild/dependencies.py

```diff
@@ -1568,6 +1568,118 @@ def __init__(self, environment, kwargs):
+    c args, cxx args, and ldargs as well as version.
+    """
+
+    # Ordered list of llvm-config binaries to try. Start with base, then try
+    # newest back to oldest (3.5 is abitrary), and finally the devel version.
+    llvm_config_bins = [
+        'llvm-config', 'llvm-config-4.0', 'llvm-config-3.9', 'llvm-config39',
+        'llvm-config-3.8', 'llvm-config38', 'llvm-config-3.7', 'llvm-config37',
+        'llvm-config-3.6', 'llvm-config36', 'llvm-config-3.5', 'llvm-config35',
+        'llvm-config-3.4', 'llvm-config34', 'llvm-config-3.3', 'llvm-config33',
+        'llvm-config-devel',
+    ]
+    llvmconfig = None
+    _llvmconfig_found = False
+    __best_found = None
+
+    # TODO: should we handle components like clang?
+    def __init__(self, environment, kwargs):
+        super().__init__('llvm-config', kwargs)
+        self.cargs = []
+        self.libs = []
+        self.components = []
+
+        req_version = kwargs.get('version', None)
+        if self.llvmconfig is None:
+            self.check_llvmconfig(req_version)
+        if not self._llvmconfig_found:
+            if self.__best_found is not None:
+                mlog.log('Dependency LLVM found:', mlog.red('NO'))
+                mlog.log('found {!r} but need:'.format(self.modversion),
+                         req_version)
+            else:
+                mlog.log("No llvm-config found; can't detect dependency")
+            return
+
+        p, out = Popen_safe([self.llvmconfig, '--version'])[:2]
+        if p.returncode != 0:
+            mlog.log('Dependency LLVM found:', mlog.red('NO'))
+        else:
+            self.modversion = out.strip()
+            mlog.log('Dependency LLVM found:', mlog.green('YES'))
+            self.is_found = True
+
+            p, out = Popen_safe([self.llvmconfig, '--libs'])[:2]
+            if p.returncode != 0:
+                raise DependencyException('Could not generate libs for LLVM.')
+            self.libs = out.split()
+
+            p, out = Popen_safe([self.llvmconfig, '--includedir'])[:2]
+            if p.returncode != 0:
+                raise DependencyException('Could not generate includedir for LLVM.')
+            self.include_directories = out.split()
+
+            p, out = Popen_safe([self.llvmconfig, '--components'])[:2]
+            if p.returncode != 0:
+                raise DependencyException('Could not generate components for LLVM.')
+            self.components = out.split()
+
+        components = kwargs.get('components')
+        if components:
+            self.validate_components(components)
+
+    def validate_components(self, components):
+        if components:
+            if isinstance(components, str):
+                if components not in self.components:
+                    mlog.log('LLVM component', components, 'found:', mlog.red('NO'))
+                    self.is_found = False
+                else:
+                    mlog.log('LLVM component', components, 'found:', mlog.green('YES'))
+
```

Comment:
These aren't unrelated blocks though, it's part of the same `if..elif..else` block, which is why I found it odd that it was there.